File: solutions/nim/allergies/2/allergies.nim
type
  Allergen* = enum
    Eggs, Peanuts, Shellfish, Strawberries, Tomatoes, Chocolate, Pollen, Cats

proc isAllergicTo*(score: int, allergen: Allergen): bool =
  (score and (1 shl ord(allergen))) != 0

proc allergies*(score: int): set[Allergen] =
  for allergen in Allergen:
    if isAllergicTo(score, allergen):
      result.incl(allergen)

File: solutions/python/allergies/5/allergies.py
class Allergies:
    def __init__(self, score):
        self.score = score

    def allergic_to(self, item):
        return item in self.lst

    @property
    def lst(self):
        allergies = dict(
            eggs=1,
            peanuts=2,
            shellfish=4,
            strawberries=8,
            tomatoes=16,
            chocolate=32,
            pollen=64,
            cats=128
        )
        return [
            allergy for allergy, val in allergies.items()
            if val & self.score
        ]
